ckanext/activityinfo/assets/js/activityinfo-download.js

```diff
@@ -0,0 +1,253 @@
+            this.container_div = this.el.querySelector('#activity_info_new_div');
+            this.dbSelect = this.el.querySelector('#ai-database-select');
+            this.formSelect = this.el.querySelector('#ai-form-select');
+            this.formatSelect = this.el.querySelector('#ai-format-select');
+            this.errorDiv = this.el.querySelector('#ai-error');
+            this.infoDiv = this.el.querySelector('#ai-info');
+            this.radioBtn = document.getElementById('resource-url-activityinfo');
+            // Include resource description to include Activity Info info
+            this.descriptionField = document.getElementById('field-description');
+            
+            // Hidden fields for form submission
+            this.dbIdField = this.el.querySelector('#ai-database-id-field');
+            this.formIdField = this.el.querySelector('#ai-form-id-field');
+            this.formatField = this.el.querySelector('#ai-format-field');
+            this.formLabelField = this.el.querySelector('#ai-form-label-field');
+            
+            this.databasesLoaded = false;
+            this.csrfToken = this.getCSRFToken();
+
+            if (!this.dbSelect) {
+                console.error('ActivityInfo: Could not find database select element');
+                return;
+            }
+
+            // Load databases when ActivityInfo option is selected
+            if (this.radioBtn) {
+                this.radioBtn.addEventListener('change', function() {
+                    if (this.checked && !self.databasesLoaded) {
+                        self.container_div.style.display = 'block';
+                        self.loadDatabases();
+                    }
+                });
+
+                var aiButton = document.getElementById('btn-activity-info');
+                if (aiButton) {
+                    aiButton.addEventListener('click', function() {
+                        if (!self.databasesLoaded) {
+                            self.container_div.style.display = 'block';
+                            setTimeout(function() {
+                                self.loadDatabases();
+                            }, 100);
+                        }
+                    });
+                }
+
+                if (this.radioBtn.checked) {
+                    this.loadDatabases();
+                }
+            }
+
+            this.dbSelect.addEventListener('change', function() {
+                self.onDatabaseChange();
+            });
+
+            this.formSelect.addEventListener('change', function() {
+                self.onFormChange();
+            });
+
+            this.formatSelect.addEventListener('change', function() {
+                self.onFormatChange();
+            });
+        },
+
+        getCSRFToken: function() {
+            var metaTag = document.querySelector('meta[name="csrf_token"]');
+            if (metaTag) return metaTag.getAttribute('content');
```

Comment:
For getting the CSRF token it should be enough to extract it from the meta tag that ships with every rendered page. 

Something like:
```js
        var csrf_field = $('meta[name=csrf_field_name]').attr('content');
        var csrf_token = $('meta[name='+ csrf_field +']').attr('content');
```

I don't feel we should be inspecting cookies. Are the other two scenarios (cookie and input) for a particular reason? 🤔